src/treebuilder.rs

```diff
@@ -0,0 +1,223 @@
+use libc::{c_int, c_void};
+use std::marker;
+use std::ffi::OsStr;
+
+use {panic, raw, Error, Oid, Repository, Tree, TreeEntry};
+use util::{Binding, IntoCString};
+
+/// Constructor for in-memory trees
+pub struct TreeBuilder<'repo> {
+    raw: *mut raw::git_treebuilder,
+    _marker: marker::PhantomData<&'repo Repository>,
+}
+
+impl<'repo> TreeBuilder<'repo> {
+    /// Create a new TreeBuilder with no entries
+    pub fn new(repo: &'repo Repository) -> Result<TreeBuilder, Error> {
```

Comment:
Currently most construction methods for types like this are on the `Repository` type itself, could this move over there with a method name like `treebuilder`?